### PR TITLE
Exception Handler

### PR DESCRIPTION
**Pull request by suiyoubi**

### CODE REVIEW

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java [hunk 1]
```diff
@@ -34,11 +36,12 @@ private String getCustomer() {
     }
 
     @PostMapping
-    private CustomerResponseDto createCustomer(@RequestBody CustomerRequestDto customerRequestDto) {
+    private CustomerResponseDto createCustomer(@RequestBody CustomerRequestDto customerRequestDto)
+            throws BadRequestException, ConflictException {
 
         if (Objects.isNull(customerRequestDto.getUsername()) ||
```
Comment (chenyirex) [inline src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: do we wanna do this check? If we want to check the validity of this request dto, we probably need to check all the `nullable = false` field right? So I think we should either have a proper validity check method that checks this dto or just don't check it at all and rely on the repository to throw an actual data integrity exception from db.

File: src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java [hunk 2]
```diff
@@ -18,12 +19,12 @@ public CustomerService(CustomerRepository customerRepository) {
...
+    public Customer createCustomer(CustomerRequestDto customerRequestDto) throws ConflictException {
 
         String username = customerRequestDto.getUsername();
 
         if (this.customerRepository.findById(username).isPresent()) {
-            // exception throws
+            throw new ConflictException("Username already exists");
         }
 
         Customer toCreateCustomer = new Customer();
```
[the remark this answers is not in the document]
Comment (chenyirex) [reply src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java]: I know it doesnt belong to this PR XD

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java (hunk 1 above)
Comment (chenyirex) [reply src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: some christophe code as reference: 
https://github.com/CareCru/oauth-client-service/blob/master/src/main/java/com/carecru/oauth/registration/client/ClientController.java
https://github.com/CareCru/oauth2-utils/blob/master/src/main/java/com/carecru/oauth2api/apiutils/dto/oauth/ClientRequestDto.java

Comment (suiyoubi) [reply src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: shoulnt belong here, in this case should be rejected by the front-end

Comment (suiyoubi) [reply src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: about your another comment, I am using postman for testing the endpoint, I am not sure what you mean by testing the controller. If you wanna write controller test that needs to mock a lot of stuffs, I feel like thats too much. We dont have enough time to finish all of them.

File: src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (suiyoubi) [reply src/main/java/com/ubc/cpsc304/hotelify/service/CustomerService.java]: okay.

File: src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java (hunk 1 above)
Comment (chenyirex) [reply src/main/java/com/ubc/cpsc304/hotelify/controller/CustomerController.java]: ok

**Pull request merged**